QuasarHelios2015/src/org/team1540/quasarhelios/AutonomousModeToteContainer.java

```diff
@@ -1,35 +1,68 @@
 package org.team1540.quasarhelios;
 
 import ccre.channel.FloatInputPoll;
+import ccre.ctrl.BooleanMixing;
 import ccre.holders.TuningContext;
 import ccre.instinct.AutonomousModeOverException;
 
-public class AutonomousModeOneTote extends AutonomousModeBase {
+public class AutonomousModeToteContainer extends AutonomousModeBase {
     protected FloatInputPoll toteDistance;
     protected FloatInputPoll autoZoneDistance;
     protected FloatInputPoll secondDistance;
     
     private FloatInputPoll containerTurnTime;
-    private FloatInputPoll containerDepositStrafeTime;
     private FloatInputPoll autoZoneAngle;
     private FloatInputPoll autoZoneSpeed;
     private FloatInputPoll nudge;
     private FloatInputPoll containerHeight;
-    private FloatInputPoll turnAngle;
 
-    public AutonomousModeOneTote() {
+    public AutonomousModeToteContainer() {
         super("One Tote");
     }
 
     @Override
     protected void runAutonomous() throws InterruptedException,
             AutonomousModeOverException {
-        // Loading
-        collectTote();
+        // Load tote.
+        
+        // Move elevator.
+        boolean elevatorAtTop = Elevator.atTop.get();
+        if (!elevatorAtTop) {
+            Elevator.setTop.event();
+        }
+
+        // Move clamp.
+        float currentClampHeight = Clamp.heightReadout.get();
+        if (currentClampHeight < AutoLoader.clampHeightThreshold.get()) {
+            Clamp.mode.set(Clamp.MODE_HEIGHT);
+            Clamp.height.set(AutoLoader.clampHeightThreshold.get());
+            if (elevatorAtTop) {
+                waitUntil(Clamp.atDesiredHeight);
+            } else {
+                waitUntil(BooleanMixing.andBooleans(Clamp.atDesiredHeight, Elevator.atTop));
+            }
+        } else {
+            if (!elevatorAtTop) {
+                waitUntil(Elevator.atTop);
+            }
+        }
```

Comment:
All the same stuff applies here. Note that you could separate the loading out into its own function.